test/pipeline_test_runner.cc

```diff
@@ -134,14 +137,15 @@ class ExpectationHandler {
         }
     }
 
-    void drainErrors() {
-        auto newErrors = errorQueue->drainAllErrors();
+    void drainErrors(const core::GlobalState &gs) {
+        errorQueue->flushAllErrors(gs);
+        auto newErrors = errorCollector->drainErrors();
         errors.insert(errors.end(), make_move_iterator(newErrors.begin()), make_move_iterator(newErrors.end()));
     }
 
-    void clear() {
+    void clear(const core::GlobalState &gs) {
         got.clear();
-        errorQueue->drainAllErrors();
+        errorQueue->flushAllErrors(gs);
```

Comment:
I think this also has to `drainErrors` on the `errorCollector` to throw the collected errors away.